Source/TradeoffAnalyticsV1/Models/Problem.swift

```diff
@@ -0,0 +1,507 @@
+        }
+    }
+}
+
+/// The direction of a column (i.e. minimize or maximize).
+public enum Goal: String {
+    
+    /// Minimize the given column.
+    case Minimize = "min"
+    
+    /// Maximize the given column.
+    case Maximize = "max"
+}
+
+/// A range of valid values for a column.
+public enum Range: JSONEncodable, JSONDecodable {
+    
+    /// High and low values that define the range of a `DateTime`
+    /// column. Valid only for `DateTime` columns.
+    case DateRange(low: NSDate, high: NSDate)
+    
+    /// High and low `Int` values that define the range of a
+    /// `Numeric` column. Valid only for `Numeric` columns.
+    case IntRange(low: Int, high: Int)
+    
+    /// High and low `Double` values that define the range of a
+    /// `Numeric` column. Valid only for `Numeric` columns.
+    case DoubleRange(low: Double, high: Double)
+    
+    /// An array of valid values that define the range of possible values
+    /// for a `Categorical` column. Valid only for `Categorical` columns.
+    case CategoricalRange(categories: [String])
+    
+    /// A date formatter to convert between `NSDate` and `String`.
+    private static let dateFormatter: NSDateFormatter = {
+        let dateFormatter = NSDateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZZZZZ"
+        return dateFormatter
+    }()
+    
+    /// Used internally to serialize a `Range` model to JSON.
+    public func toJSON() -> JSON {
+        var json = [String: JSON]()
+        switch self {
+        case .DateRange(let low, let high):
+            json["low"] = .String(Range.dateFormatter.stringFromDate(low))
+            json["high"] = .String(Range.dateFormatter.stringFromDate(high))
+            return .Dictionary(json)
+        case .IntRange(let low, let high):
+            json["low"] = .Int(low)
+            json["high"] = .Int(high)
+            return .Dictionary(json)
+        case .DoubleRange(let low, let high):
+            json["low"] = .Double(low)
+            json["high"] = .Double(high)
+            return .Dictionary(json)
+        case .CategoricalRange(let categories):
+            return .Array(categories.map { .String($0) })
+        }
+    }
+    
+    /// Used internally to initialize a `Range` model from JSON.
+    public init(json: JSON) throws {
+        // try to parse as `Range.DateRange`
+        if let low = try? json.string("low"), high = try? json.string("high") {
+            let lowDate = Range.dateFormatter.dateFromString(low)
+            let highDate = Range.dateFormatter.dateFromString(high)
+            if let lowDate = lowDate, highDate = highDate {
+                self = .DateRange(low: lowDate, high: highDate)
+                return
+            }
+        }
+        
+        // try to parse as `Range.IntRange`
+        if let low = try? json.int("low"), high = try? json.int("high") {
+            self = .IntRange(low: low, high: high)
+            return
+        }
+        
```

Comment:
Investigate whether this will truncate a `double` into an `int`, essentially bypassing the possibility to return a `double`